pkg/plugins/runtime/k8s/controllers/gatewayapi/common/reconcile.go

```diff
@@ -81,11 +90,20 @@ func ReconcileLabelledObject(
 	}
 
 	if existing != nil {
+		existingSpec, err := existing.GetSpec()
+		if err != nil {
+			return err
+		}
+		if proto.Equal(existingSpec, ownedSpec) {
```

Comment:
kind of sneaking an unrelated change in here :wink: 

This happens all over the place. And it seems error prone, what if the meta has changed? Can we do this instead with `client.Patch`?

Reply:
Do you mean that meta can change in other places? From what I'm reading, `ReconcileLabelledObject` retrieves an existing object and updates the spec.

Reply:
I just meant it's fragile. If logic is ever introduced earlier on that does change meta, this would be broken. 

Of course this is only one spot but before we add `proto.Equal` guards inline before k8s `Update` calls everywhere we can probably as a rule use a helper that works with `Patch` and skip the API call if the patch is empty. I'll look into it

Reply:
Discussed offline, we'll merge this for now and come up with a helper if this pattern is necessary elsewhere.

Reply:
Talked offline, we decided to leave it for now as this is a helper already. If we see more occurrences of such logic, we can try to extract this to a helper.